template_handler/cdk.py

```diff
@@ -0,0 +1,199 @@
+
+        Sceptre can assume roles, profiles, etc... to connect to AWS for a given stack. This is
+        very useful. However, we need that SAME connection information to carry over to SAM when we
+        invoke it. The most precise way to do this is to use the same session credentials being used
+        by Sceptre for other stack operations. This method obtains those credentials and sets them
+        as environment variables that are passed to the subprocess and will, in turn, be used by
+        SAM CLI.
+
+        The environment variables dict created by this method will inherit all existing
+        environment variables in the current environment, but the AWS connection environment
+        variables will be overridden by the ones for this stack.
+
+        Returns:
+            The dictionary of environment variables.
+        """
+        envs = os.environ
+        # Set aws environment variables specific to whatever AWS configuration has been set on the
+        # stack's connection manager.
+        credentials: Credentials = self.connection_manager._get_session(
+            self.connection_manager.profile,
+            self.connection_manager.region,
+            self.connection_manager.iam_role
+        ).get_credentials()
+        envs.update(
+            AWS_ACCESS_KEY_ID=credentials.access_key,
+            AWS_SECRET_ACCESS_KEY=credentials.secret_key,
+        )
+
+        # There might not be a session token, so if there isn't one, make sure it doesn't exist in
+        # the envs being passed to the subprocess
+        if credentials.token is None:
+            envs.pop('AWS_SESSION_TOKEN', None)
+        else:
+            envs['AWS_SESSION_TOKEN'] = credentials.token
+
+        return envs
+
+    def _cmd_exists(self, cmd) -> bool:
+        """
+        Checks whether a specified CLI command exists
+        """
+
+        cmd_exists = shutil.which(cmd) is not None
+        self.logger.debug(f"command '{cmd}' exists: {cmd_exists}")
+        return cmd_exists
+
+    def _node_package_exists(self, package) -> bool:
+        """
+        Checks whether a specifie Node package exists either in the workspace or global scope
+        """
+
+        package_exists = False
+        workspace_result = subprocess.run(
+            f'npm list {package}',
+            shell=True,
+            capture_output=True
+        )
+        self.logger.debug(f"Workspace NPM package '{package}' exists: {not bool(workspace_result.returncode)}")
+
+        if workspace_result.returncode == 0:
+            package_exists = True
+        else:
+            global_result = subprocess.run(
+                f'npm --global list {package}',
+                shell=True,
+                capture_output=True
+            )
+            self.logger.debug(f"Global NPM package '{package}' exists: {not bool(global_result.returncode)}")
+
+            if global_result.returncode == 0:
+                package_exists = True
+
+        return package_exists
```

Comment:
This might be a bit overkill. If you run `subprocess` with `check=True`, any nonzero exit code will result in an exception being raised. In that case, I think it would just make sense to do the thing you want to do and let it blow up with a relevant exception if it can't instead of doing all this checking beforehand.

Reply:
You're right. It's not a big deal.